profiles/lib/relations.lua

```diff
@@ -0,0 +1,100 @@
+-- Profile functions dealing with various aspects of relation parsing
+--
+-- You can run a selection you find useful in your profile,
+-- or do you own processing if/when required.
+
+Utils = require('lib/utils')
+
+Relations = {}
+
+function Relations.Merge(relations)
+  local result = {}
+
+  for _, r in ipairs(relations) do
+    for k, v in pairs(r) do
+      result[k] = v
+    end
+  end
+
+  return result
+end
+
+-- match ref values to relations data
+function Relations.MatchToRef(relations, ref)
+
+  function calculate_scores(refs, tag_value)
+    local tag_tokens = Set(Utils.tokenize_common(tag_value))
+    local result = {}
+    for i, r in ipairs(refs) do
+      local ref_tokens = Utils.tokenize_common(r)
+      local score = 0
+
+      for _, t in ipairs(ref_tokens) do
+        if tag_tokens[t] then
+          if Utils.is_number(t) then
```

Comment:
It’s common for routes to have alphabetic suffixes, for example [`ref=I 35W` on a way](http://www.openstreetmap.org/way/50715899) or [`ref=35W`](http://www.openstreetmap.org/relation/2145877) on a (northbound) route relation.

Reply:
Can't find any example, where current algorithm will work incorrect in this case.

Reply:
After rereading this code, I think the suffix case will be a nonissue. However, for [this way](http://www.openstreetmap.org/way/122502126), wouldn’t the two items in the way `ref` (`PA 88` and `PA 88 Truck`) tie because `88` appears in both items and both relations’ `ref` tags are set to `88`? We could factor in the relations’ `network` tags, sort of like what’s described in https://github.com/Project-OSRM/osrm-backend/pull/4478#discussion_r138947028.